### PR TITLE
adds possibility to load jar with bundles form BUNDLES_JARS_PATH directory

### PR DESCRIPTION
**Pull request by sidorovis**
adds possibility to load jar files with bundles from ConfigurationService.PNAME_SC_HOME_DIR_LOCATION / ConfigurationService.PNAME_SC_HOME_DIR_NAME / 
bundles/
directory

### CODE REVIEW

**Comment by bgrozev** [comment]
Jenkins, add to whitelist

**Comment by bgrozev** [comment]
This looks like a legit failure:
http://pr-reports.jitsi.org/jicoco-PR-test/23/log.txt

**Comment by sidorovis** [comment]
Hm, it seems that jicoco tries to use previous version of jitsi-android-osgi. 
(Exception in thread "main" java.lang.NoSuchMethodError: org.jitsi.impl.osgi.framework.OSGiLauncher.<init>([[Ljava/lang/String;Ljava/lang/ClassLoader;)V)

Could I get this testing script to understand some details?

**Comment by damencho** [comment]
@sidorovis So PR testing creates a deployment running from source. We don't have the scripts somewhere publicly available at the moment.
But what it does it runs everything from source except for example jicoco, it uses the source from your PR branch named 'adds_class_loader_for_bundles'. If it finds a PR opened against any of the other repositories using the same branch name it will use those for it.
What I see is that jicofo and jvb both use the wrong version of android-osgi (jitsi-android-osgi-1.0-20150723.002702-1).
So what you need to do is open two new PRs one in jvb and one in jicofo using the same branch name, where you update the jitsi-android-osgi, and that way the 3 PRs will be tested all together and  hopefully the tests should pass.
Hope this helps.

**Comment by sidorovis** [comment]
Update versions of jitsi-android-osgi is OK. 
Should I update jicoco version at pom.xml of JVB, Jicofo somehow or this two PR's will wait till jicoco release, receive new jicoco versions afterwards and be merged afterwards?
Thanks!

**Comment by bgrozev** [comment]
Step 1: open PRs with the same branch name which update jitsi-android-osgi only. This will result in the tests being run with the specified jitsi-android-osgi version, and with SNAPSHOT versions of the projects with the same branch name (jicoco, jicoco, jitsi-videobridge).
Step 2: We merge this PR.
Step 3: Update the PRs in jitsi-videobridge and jicofo with the new version of jicoco (which will be published to the maven repo automatically after the merge)
Step 4: Merge the other 2 PRs 

Thanks for sticking with all this :)

**Comment by sidorovis** [comment]
Related push requests.
https://github.com/jitsi/jitsi-videobridge/pull/632
https://github.com/jitsi/jicofo/pull/268

BTW: for some reason current version of pom.xml jicofo depends from jicoco 1.1-SNAPSHOT version.

**Comment by sidorovis** [comment]
Passed by https://github.com/jitsi/jitsi-videobridge/pull/632 (http://pr-reports.jitsi.org/jitsi-videobridge-PR-test/724/).

**Comment by sidorovis** [comment]
@damencho added javadocs / license. Again about jicofo: for some reason jicofo pom.xml as jicoco dependency version has 1.1-SNAPSHOT. I believe that is a mistake. After the merge, should I update with "release" version instead?

**Comment by damencho** [comment]
Yes, jicofo should be updated with exact version. This is a mistake, right.

**Comment by sidorovis** [comment]
@damencho is it still require changes?

**Comment by damencho** [comment]
Merged, the result artifact is jicoco-1.1-20180327.183532-3

File: src/main/java/org/jitsi/meet/ComponentMain.java [hunk 1]
```diff
@@ -182,12 +184,28 @@ private ClassLoader loadBundlesJars(OSGiBundleConfig bundleConfig) {
         String bundlesJarsPath = bundleConfig.getBundlesJarsPath();
         if (bundlesJarsPath == null)
         {
-            return ClassLoader.getSystemClassLoader();
```
Comment (damencho) [inline src/main/java/org/jitsi/meet/ComponentMain.java]: @sidorovis Why do we need this?

I was testing running our component on java 11 and this change using the class loader returned by getPlatformClassLoader breaks them.

Comment (damencho) [reply src/main/java/org/jitsi/meet/ComponentMain.java]: I'm currently testing and plan to remove it.

Comment (damencho) [reply src/main/java/org/jitsi/meet/ComponentMain.java]: I see in the comments of https://github.com/jitsi/jitsi-android-osgi/pull/1 it was a discussion with @ibauersachs 
In jitsi-android-osgi we want the classloader that has access to the whole classpath, which is not the case for Felix.  Our lib is just a wrapper so we can reuse and run the OSGi bundles in a normal classLoader way ... 
I just tested and everything works with just ClassLoader.getSystemClassLoader() when using java11. There are few more minor modifications and I will be creating a PR soon.

**Pull request merged**